src/com/phonebookservice/server/FileDataAccessAdapter.java

```diff
@@ -1,13 +1,56 @@
 package com.phonebookservice.server;
 
+import java.io.FileNotFoundException;
+
+import com.phonebookservice.config.Config;
+import com.phonebookservice.config.Config.ConfigKey;
+import com.phonebookservice.database.ContactsDatabase;
+import com.phonebookservice.exception.InternalServerException;
 import com.phonebookservice.model.Contact;
+import com.phonebookservice.util.StringUtility;
 
 /**
  * file data access adapter.
  *
  * @author Marwa Saleh
  */
-public class FileDataAccessAdapter implements IDataAccessAdapter<Contact> {
+public final class FileDataAccessAdapter
+        implements IDataAccessAdapter<Contact> {
+
+    private static FileDataAccessAdapter singleton;
+    private final ContactsDatabase contactsDb;
+
+    /**
+     * Initialization of file data access adapter.
+     *
+     * @param databaseLink the database link.
+     */
+    private FileDataAccessAdapter(final String databaseLink) {
+        if (StringUtility.isNullOrEmptyString(databaseLink)
```

Comment:
If user send to wrong data, it should be bad request not internal server.

This is client error not a service error. Right ?

Reply:
yes you are right